### ISSUE TITLE
TeachingPopoverSurface does not have border in high windows contrast mode

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#30929]
This is the Popover in windows high contrast mode
![image](https://github.com/microsoft/fluentui/assets/20744592/0570e108-f82a-4857-a7a6-9939b4c14637)

This is the TeachingPopover in windows high contrast mode
![image](https://github.com/microsoft/fluentui/assets/20744592/3525b820-aa53-4f77-9be3-d2cbdf4c3182)
